benchmarks/runner/run_benchmarks.py

```diff
@@ -0,0 +1,297 @@
+
+import csv
+import subprocess
+import json
+import os
+import datetime
+
+
+def run_mhp(
+    vec_size: int,
+    reps: int,
+    filter: str,
+    sycl_used: bool,
+    only_fsycl: bool,
+    mpirun: bool = True,
+    n: int = None,
+    pin_domain: str = None,
+    pin_order: str = None,
+    pin_cell: str = None,
+):
+    os.environ["BENCHMARK_FILTER"] = filter
+    if pin_domain is not None:
+        os.environ["I_MPI_PIN_DOMAIN"] = pin_domain
+    if pin_order is not None:
+        os.environ["I_MPI_PIN_ORDER"] = pin_order
+    if pin_cell is not None:
+        os.environ["I_MPI_PIN_CELL"] = pin_cell
+    if only_fsycl:
+        bench_path = "./build/benchmarks/gbench/mhp/mhp-bench-only-fsycl"
+    else:
+        bench_path = "./build/benchmarks/gbench/mhp/mhp-bench"
+
+    time_now_string = datetime.datetime.now().isoformat(timespec='minutes')
+    directory_path = f"build/benchmarks/json_output/mhp"
+    if not os.path.exists(directory_path):
+        os.makedirs(directory_path)
+    bench_out = f"--benchmark_out={directory_path}/mhp_benchmark_{time_now_string}.json --benchmark_out_format=json"
+    if n is None:
+        n = ""
+    else:
+        n = f"-n {str(n)}"
+    if mpirun:
+        mpirun = "mpirun"
+    else:
+        mpirun = ""
+        n = ""
+    if sycl_used:
+        command = [
+            mpirun,
+            n,
+            bench_path,
+            "--sycl",
+            "--vector-size",
+            str(vec_size),
+            "--reps",
+            str(reps),
+            bench_out,
+        ]
+    else:
+        command = [
+            mpirun,
+            n,
+            bench_path,
+            "--vector-size",
+            str(vec_size),
+            "--reps",
+            str(reps),
+            bench_out,
+        ]
+    while "" in command:
+        command.remove('')
+    command = " ".join(command)
+    process = subprocess.Popen(
+        command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True
+    )
+    output, err = process.communicate()
+    print(output)
+    print(err)
+
+
+def run_shp(
+    vec_size: int,
+    reps: int,
+    d: int,
+    only_fsycl: bool,
+    filter: str = None,
+    kmp_aff: str = None,
+):
+    if filter is not None:
+        os.environ["BENCHMARK_FILTER"] = filter
+    else:
+        try:
+            os.unsetenv("BENCHMARK_FILTER")
+        except:
+            pass
+    if kmp_aff is not None:
+        os.environ["KMP_AFFINITY"] = kmp_aff
+    if only_fsycl:
+        bench_path = "./build/benchmarks/gbench/shp/shp-bench-only-fsycl"
+    else:
+        bench_path = "./build/benchmarks/gbench/shp/shp-bench"
+    time_now_string = datetime.datetime.now().isoformat(timespec='minutes')
+    directory_path = f"build/benchmarks/json_output/shp"
+    if not os.path.exists(directory_path):
+        os.makedirs(directory_path)
+    bench_out = f"--benchmark_out={directory_path}/shp_benchmark_{time_now_string}.json --benchmark_out_format=json"
```

Comment:
I made that decision because csv reporter is deprecated (https://github.com/google/benchmark/blob/main/docs/user_guide.md#output-files). Also I talked with Lukasz about converting json to csv but we decided to leave it as it is for now

Reply:
ok